File: Projects/Projects_vb/ZengFanyun/cmds_test.vb
Imports Autodesk.Revit.ApplicationServices
Imports Autodesk.Revit.DB.Events

Namespace OldW.Commands

    <Autodesk.Revit.Attributes.Transaction(Autodesk.Revit.Attributes.TransactionMode.Manual)>
    Public Class cmd_Test : Implements IExternalCommand
        Private WithEvents uiApp As UIApplication
        Private uiDoc As UIDocument
        Private WithEvents App As Application

        Public Function Execute(commandData As ExternalCommandData, ByRef message As String, elements As ElementSet) As Result Implements IExternalCommand.Execute
            App = commandData.Application.Application
            uiDoc = commandData.Application.ActiveUIDocument

            Dim a = uiDoc.Selection.PickObjects(objectType:=Selection.ObjectType.Element)
            MessageBox.Show(a.Count)
            Return Result.Cancelled
        End Function


    End Class
End Namespace

File: Projects/Projects/ZengFanyun/cmds_test.cs
// VBConversions Note: VB project level imports

using System.Windows.Forms;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;
using Autodesk.Revit.UI.Selection;
using Application = Autodesk.Revit.ApplicationServices.Application;

//using Autodesk.Revit.DB;
//using Autodesk.Revit.UI;

// End of VB project level imports

//using Autodesk.Revit.ApplicationServices;
//using Autodesk.Revit.DB.Events;


namespace OldW.Commands
		{
			
			[Autodesk.Revit.Attributes.Transaction(Autodesk.Revit.Attributes.TransactionMode.Manual)]public class cmd_Test : IExternalCommand
				{
				private UIApplication uiApp;
		private UIDocument uiDoc;
		private Application App;
		
		public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
		{
			App = commandData.Application.Application;
			uiDoc = commandData.Application.ActiveUIDocument;
			
			var a = uiDoc.Selection.PickObjects(ObjectType.Element);
			MessageBox.Show(System.Convert.ToString(a.Count));
			return Result.Cancelled;
		}
		
		
	}
}

